components/Dialogs/src/Theming/UIViewController+MaterialTheming.m

```diff
@@ -12,24 +12,28 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#import "MDCDialogPresentationController+MaterialTheming.h"
+#import "UIViewController+MaterialTheming.h"
 
 static const CGFloat kCornerRadius = 4;
 
-@implementation MDCDialogPresentationController (MaterialTheming)
+@implementation UIViewController (MaterialTheming)
 
-- (void)applyThemeWithScheme:(nonnull id<MDCContainerScheming>)scheme {
+- (void)applyPresentationThemeWithScheme:(nonnull id<MDCContainerScheming>)scheme {
   // Color
   id<MDCColorScheming> colorScheme = scheme.colorScheme;
   if (!colorScheme) {
     colorScheme =
         [[MDCSemanticColorScheme alloc] initWithDefaults:MDCColorSchemeDefaultsMaterial201804];
   }
-  self.scrimColor = [colorScheme.onSurfaceColor colorWithAlphaComponent:(CGFloat)0.32];
+  MDCDialogPresentationController *presentationController = self.mdc_dialogPresentationController;
```

Comment:
I don't think this handles the case where a view controller is themed before it is presented.  Won't self.mdc_dialogPresentationController be nil?

Reply:
self.mdc_dialogPresentationController gets initialized first time it's referenced, if the transitionController is set. so it is not nil, even before the it is presented.
Also - I abandoned this implementation in favor of solution C (see in description) - going back to old theming extension on MDCDialogPresentationController.